File: backend/internal/model/login_log.go
package model

import "time"

type LoginLog struct {
	SnowflakeModel
	UserID        int64      `json:"user_id" gorm:"index"`
	Username      string     `json:"username" gorm:"size:100;index"`
	TenantID      int64      `json:"tenant_id" gorm:"index"`
	LoginStatus   string     `json:"login_status" gorm:"size:20;index;not null"`
	LoginIP       string     `json:"login_ip" gorm:"size:128"`
	LoginLocation string     `json:"login_location" gorm:"size:255"`
	Browser       string     `json:"browser" gorm:"size:200"`
	OS            string     `json:"os" gorm:"size:200"`
	LoginTime     time.Time  `json:"login_time" gorm:"index"`
	LogoutTime    *time.Time `json:"logout_time"`
	Msg           string     `json:"msg" gorm:"size:500"`
}

func (LoginLog) TableName() string {
	return "login_logs"
}

File: web/src/api/loginLog.ts
import request, { type PageResult } from './request'

export interface LoginLog {
  id: number
  user_id: number
  username: string
  tenant_id: number
  login_status: string
  login_ip: string
  login_location: string
  browser: string
  os: string
  login_time: string
  logout_time: string | null
  msg: string
}

export interface LoginLogParams {
  page?: number
  page_size?: number
  username?: string
  login_status?: string
  login_ip?: string
  start_time?: string
  end_time?: string
}

const loginLogApi = {
  listLogs(params?: LoginLogParams) {
    return request.get<any, { code: number; data: PageResult<LoginLog> }>('/login-logs', { params })
  },

  cleanup(days: number) {
    return request.post('/login-logs/cleanup', null, { params: { days } })
  },
}

export default loginLogApi
